### PR TITLE
update API: is_xxx to check_xxx when self is mutated

### PR DESCRIPTION
**Pull request by zhenfeizhang**
## Description

closes: #56 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] ~Wrote unit tests~
- [x] Updated relevant documentation in the code
- [ ] ~Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`~
- [x] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

File: plonk/src/circuit/customized/mod.rs [hunk 1]
```diff
@@ -964,7 +964,7 @@ pub(crate) mod test {
     fn build_is_zero_circuit<F: PrimeField>(a: F) -> Result<PlonkCircuit<F>, PlonkError> {
```
Comment (alxiong) [inline plonk/src/circuit/customized/mod.rs:964]: we should change these test fn names as well.

File: plonk/src/circuit/customized/mod.rs [hunk 2]
```diff
@@ -1394,7 +1394,7 @@ pub(crate) mod test {
     fn build_is_in_range_circuit<F: PrimeField>(a: F) -> Result<PlonkCircuit<F>, PlonkError> {
```
Comment (alxiong) [inline plonk/src/circuit/customized/mod.rs:1394]: same here. update fn name

File: plonk/src/circuit/customized/mod.rs (hunk 1 above)
Comment (zhenfeizhang) [reply plonk/src/circuit/customized/mod.rs]: nice catch. Fixed.

**Review by alxiong — APPROVED** [review]
LGTM

**Pull request merged**